cmd/destroy/destroy.go

```diff
@@ -167,12 +169,14 @@ func Destroy(ctx context.Context) *cobra.Command {
 				k8sClientProvider: okteto.NewK8sClientProvider(),
 			}
 
-			kubeconfigPath := deploy.GetTempKubeConfigFile(name)
+			kubeconfigPath := getTempKubeConfigFile(name)
 			if err := kubeconfig.Write(okteto.Context().Cfg, kubeconfigPath); err != nil {
 				return err
 			}
 			os.Setenv("KUBECONFIG", kubeconfigPath)
-			defer os.Remove(kubeconfigPath)
+			if !utils.LoadBoolean(model.OktetoWithinDeployCommandContextEnvVar) {
```

Comment:
This check is spread around all the code and it is error prone. We might forget to include it some where in the future.

I think we should create an abstraction and have 2 command runners. One to be used when `model.OktetoWithinDeployCommandContextEnvVar` is true and another one to be used when it is false.

So it is clearer what is the behavior in both cases. Some functions can be common but other can be defined per runner. For example, `setErrorStatus` looks a good candidate to be defined per runner. A new `DestroyConfig` function can also be a good candidate. 

In this way, we only have this check once (when creating the runner) and it is easier to see what is executed on each type

Reply:
Thanks for the tip, I'll refactor the code with those changes 😄